### ISSUE TITLE
TypeError: Cannot read property 'format' of undefined after multiple failed sells

### ISSUE THREAD

**Issue by GuyPaddock** [askmike/gekko#1042]
* **I'm submitting a ...**
[x] bug report
[ ] feature request
[ ] question about the decisions made in the repository

* **Action taken** (what you did)
# Left Gekko running overnight, trading on GDAX.

* **Expected result** (what you hoped would happen)
  - Trades complete without error.
  - Gekko continues running until stopped.

* **Actual result** (unexpected outcome)
  - Trades get stuck in a retry loop because amount is too small.
  - Gekko eventually crashes.

* **Other information**
Apologies if this is possibly already fixed; I'm running `develop` at 43e10e1e93221a112314017ab05f354acf57e87a and have been since #980 was closed, to verify that GDAX is happy now. It has been running steadily along without any issue until this morning's crash.

Here's an abbreviated log:
```
2017-08-25 01:02:05 (INFO):     We have new trading advice!
2017-08-25 01:02:05 (INFO):              Position: short
2017-08-25 01:02:05 (INFO):              Market price: 4315.63
2017-08-25 01:02:05 (INFO):              Based on market time: 2017-08-25 05:01:00

2017-08-25 01:02:05 (INFO):     Trader Received advice to go short. Selling  BTC
2017-08-25 01:02:10 (INFO):     Attempting to SELL 0.01104053 BTC at GDAX price: 4315.64
2017-08-25 01:02:12 (ERROR):    Error buying at GDAX: { message: 'Order size is too small. Minimum size is 0.01' }
2017-08-25 01:02:12 (DEBUG):    GDAX returned an error, retrying..
...
2017-08-25 01:05:24 (DEBUG):    Requested BTC/USD trade data from GDAX ...
2017-08-25 01:05:25 (DEBUG):    Processing 2 new trades. From 2017-08-25 05:05:11 UTC to 2017-08-25 05:05:21 UTC. (a few seconds)
2017-08-25 01:05:28 (ERROR):    Error buying at GDAX: { message: 'Order size is too small. Minimum size is 0.01' }
2017-08-25 01:05:28 (DEBUG):    GDAX returned an error, retrying..
2017-08-25 01:05:33 (INFO):     SELL was successfull
2017-08-25 01:05:34 (DEBUG):    GDAX NotFound error, spoofing order
2017-08-25 01:05:34 (DEBUG):    GDAX NotFound error, spoofing order
2017-08-25 01:05:35 (INFO):     (PROFIT REPORT) original simulated balance:      82.93900318 USD
2017-08-25 01:05:35 (INFO):     (PROFIT REPORT) current simulated balance:       82.06454243 USD
2017-08-25 01:05:35 (INFO):     (PROFIT REPORT) simulated profit:                -0.87446075 USD (-1.05434201%)
2017-08-25 01:05:35 (INFO):     (ROUNDTRIP) entry date          exit date               exposed duration P&L                    profit
/opt/teamcity/gekko/plugins/performanceAnalyzer/logger.js:52
    rt.entryAt.format('YYYY-MM-DD HH:mm'),
              ^

TypeError: Cannot read property 'format' of undefined
    at Logger.logRoundtrip (/opt/teamcity/gekko/plugins/performanceAnalyzer/logger.js:52:15)
    at Logger.handleRoundtrip (/opt/teamcity/gekko/plugins/performanceAnalyzer/logger.js:131:10)
    at PerformanceAnalyzer.handleRoundtrip (/opt/teamcity/gekko/plugins/performanceAnalyzer/performanceAnalyzer.js:119:16)
    at PerformanceAnalyzer.bound [as handleRoundtrip] (/opt/teamcity/gekko/node_modules/lodash/dist/lodash.js:729:21)
    at PerformanceAnalyzer.logRoundtripPart (/opt/teamcity/gekko/plugins/performanceAnalyzer/performanceAnalyzer.js:94:10)
    at PerformanceAnalyzer.bound [as logRoundtripPart] (/opt/teamcity/gekko/node_modules/lodash/dist/lodash.js:729:21)
    at PerformanceAnalyzer.processTrade (/opt/teamcity/gekko/plugins/performanceAnalyzer/performanceAnalyzer.js:72:8)
    at Trader.bound (/opt/teamcity/gekko/node_modules/lodash/dist/lodash.js:729:21)
    at emitOne (events.js:96:13)
    at Trader.emit (events.js:188:7)
```

**Comment by askmike** [comment]
What trader is enabled in your config?

**Comment by GuyPaddock** [comment]
Full trading (not paper trading).

```
config.trader = {
  enabled: true,
  key: 'MY_KEY',
  secret: 'MY_SECRET',
  passphrase: 'MY_PASSPHRASE'
}
```

**Comment by askmike** [comment]
Ah I see what is going on: GDAX couldn't find any of the orders Gekko used. I'll let gdax run for a long time to try and figure this out..

**Comment by GuyPaddock** [comment]
For the logs it looks like the SELL failed, with a retry, for long enough that a BUY tried going through while it was still in the retries for SELL (I cut out a lot of retries from the log, so the pattern might not be as obvious, but it's there).

**Comment by GuyPaddock** [comment]
Possibly related, or a separate bug...
```
2017-08-25 17:13:17 (DEBUG):    calculated MACD properties for candle:
2017-08-25 17:13:17 (DEBUG):             short: 4335.50722879
2017-08-25 17:13:17 (DEBUG):             long: 4344.04584815
2017-08-25 17:13:17 (DEBUG):             macd: -8.53861936
2017-08-25 17:13:17 (DEBUG):             signal: -9.05846831
2017-08-25 17:13:17 (DEBUG):             macdiff: 0.51984895
2017-08-25 17:13:17 (DEBUG):    In uptrend since 1 candle(s)

2017-08-25 17:13:17 (INFO):     We have new trading advice!
2017-08-25 17:13:17 (INFO):              Position: long
2017-08-25 17:13:17 (INFO):              Market price: 4335.99
2017-08-25 17:13:17 (INFO):              Based on market time: 2017-08-25 21:12:00

2017-08-25 17:13:17 (INFO):     Trader Received advice to go long. Buying  BTC
2017-08-25 17:13:18 (INFO):     Attempting to BUY 0.01883992135415724 BTC at GDAX price: 4335.97999999
2017-08-25 17:13:37 (DEBUG):    Requested BTC/USD trade data from GDAX ...
2017-08-25 17:13:37 (DEBUG):    Processing 9 new trades. From 2017-08-25 21:13:19 UTC to 2017-08-25 21:13:32 UTC. (a few seconds)
2017-08-25 17:13:57 (DEBUG):    Requested BTC/USD trade data from GDAX ...
2017-08-25 17:13:57 (DEBUG):    Processing 3 new trades. From 2017-08-25 21:13:42 UTC to 2017-08-25 21:13:52 UTC. (a few seconds)
2017-08-25 17:14:17 (DEBUG):    Requested BTC/USD trade data from GDAX ...
2017-08-25 17:14:17 (DEBUG):    Processing 13 new trades. From 2017-08-25 21:13:59 UTC to 2017-08-25 21:14:12 UTC. (a few seconds)
2017-08-25 17:14:23 (INFO):     BUY order was not (fully) filled, cancelling and creating new order
2017-08-25 17:14:23 (DEBUG):    Waiting 10 seconds before starting a new trade on GDAX!
2017-08-25 17:14:34 (ERROR):    Wanted to buy BTC but the amount is too small (0.000000088254) at GDAX
2017-08-25 17:14:37 (DEBUG):    Requested BTC/USD trade data from GDAX ...
2017-08-25 17:14:45 (DEBUG):    Processing 3 new trades. From 2017-08-25 21:14:18 UTC to 2017-08-25 21:14:18 UTC. (a few seconds)
2017-08-25 17:14:57 (DEBUG):    Requested BTC/USD trade data from GDAX ...
2017-08-25 17:15:05 (DEBUG):    Processing 1 new trades. From 2017-08-25 21:14:22 UTC to 2017-08-25 21:14:22 UTC. (a few seconds)
2017-08-25 17:15:17 (DEBUG):    Requested BTC/USD trade data from GDAX ...
2017-08-25 17:15:37 (DEBUG):    Requested BTC/USD trade data from GDAX ...
2017-08-25 17:15:39 (DEBUG):    Processing 5 new trades. From 2017-08-25 21:14:27 UTC to 2017-08-25 21:15:38 UTC. (a minute)
2017-08-25 17:15:39 (DEBUG):    No new trades.
```

The account currently has $81.68 available with no open orders, but for some reason Gekko could not properly put through the trade, saying that the amount was too small.
![image](https://user-images.githubusercontent.com/2631799/29734260-2312497c-89bf-11e7-98cb-f09f2bea1a50.png)

**Comment by askmike** [comment]
That's another issue. But are you 100% sure you've used API keys for that account? Because Gekko is quite specific about the `0.000000088254` USD..

**Comment by GuyPaddock** [comment]
Yup, same API key the whole time. It's trading on it right now.
